syntax = "proto3";
package gateway;
import "ipss.proto";

message Empty {}

message TokenMessage {
    string token = 1;
}

message Instance {
    ipss.Instance instance = 1;
    string token = 2;
}

message ServiceTransport {
    oneof oneOf{
        string hash = 1;
        ipss.Service service = 2;
    }
    optional ipss.Configuration config = 3;
}

service Gateway {
    rpc StartService( stream ServiceTransport) returns (Instance) {}
    rpc StopService( TokenMessage ) returns (Empty) {}
}